adding expansions to add up baskets and catch weight

melinashak-noaa opened the pull request:
Implemented one expansion that adds up catch and basket weight. I'll continue to work on additional ones with actual equations, just wanted to get this out first to make sure I'm on the right track.

wsmith-nwfsc commented on bn-expansions/package.json:
```diff
@@ -1,27 +1,27 @@
-    },
-    "dependencies": {
-      "geojson": "^0.5.0",
-      "davenport": "^2.8.2",
```
might consider removing davenport if you're not using it.

melinashak-noaa replied:
okay removed it

wsmith-nwfsc commented on bn-expansions/src/calculateWeight.ts:
```diff
@@ -0,0 +1,53 @@
+import { BaseCatch } from '@boatnet/bn-models';
+
+export function updateCatchWeight(weightMethod: number, catchVal: BaseCatch) {
+    if (weightMethod === 3) {
```
could consider a switch statement instead of series of if/ else statements. If we have fallthrough cases (e.g. several weight methods using the same calculations) then a switch statement would be a little cleaner.

melinashak-noaa replied:
ahhh good point updated to switch statement also created an enum to map numbers to weight methods

wsmith-nwfsc commented on bn-expansions/src/calculateWeight.ts:
```diff
@@ -0,0 +1,53 @@
+}
+
+// updates baskets and species weights
+function getWholeWeight(catchVal: BaseCatch) {
```
Looks good to me!
I think we will want to test this via some automated tests, to verify.
Take a look at boatnet-module/test-boatnet-module/tests for examples of this. After you yarn install, yarn test:unit to see how they work.
Maybe we could add a testdata directory with .json files (stripped of any PII) with stuff to test.

melinashak-noaa replied:
okay added a unit test

melinashak-noaa commented:
okay addressed your comments and also added in a new unit test and updated the model with some expansions stuff. let me know what you think. Also pulled your changes into this review making it a bit confusing as to what changes I have made if it's too confusing let me know and I can try creating a new PR.
